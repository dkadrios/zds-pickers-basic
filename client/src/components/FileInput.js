import PropTypes from 'prop-types'

const adaptFileEventToValue = delegate => e => delegate(e.target.files[0])

const FileInput = ({
  input: { value: omitValue, onChange, onBlur, ...inputProps },
  meta: { touched, error, warning },
  ...props
}) => (
  <div>
    <input
      accept="text/plain"
      onBlur={adaptFileEventToValue(onBlur)}
      onChange={adaptFileEventToValue(onChange)}
      type="file"
      {...inputProps}
      {...props}
    />
    <p>{touched ? error || warning : ''}</p>
  </div>
)

FileInput.propTypes = {
  input: PropTypes.object.isRequired, // eslint-disable-line
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
    warning: PropTypes.string,
  }).isRequired,
}

export default FileInput
