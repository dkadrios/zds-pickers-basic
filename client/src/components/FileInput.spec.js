import React from 'react'
import flow from 'lodash/fp/flow'
import set from 'lodash/fp/set'
import FileInput from './FileInput'

const onChange = jest.fn()
const onBlur = jest.fn()

const input = { value: '', onChange, onBlur }
const meta = { touched: false, error: '', warning: '' }

const props = {
  input,
  meta,
}

describe('FileInput tests', () => {
  it('renders correctly', () => {
    shallowExpect(FileInput)(props).toMatchSnapshot()
  })

  it('should render an error when touched', () => {
    const localProps = flow(
      set('meta.error', 'error'),
      set('meta.touched', true),
    )(props)
    shallowExpect(FileInput)(localProps).toMatchSnapshot()
  })

  it('should render a warning when touched', () => {
    const localProps = flow(
      set('meta.warning', 'warning'),
      set('meta.touched', true),
    )(props)
    shallowExpect(FileInput)(localProps).toMatchSnapshot()
  })

  it('should fire onChange', () => {
    const filename = '/a/path'
    const wrapper = shallow(<FileInput {...props} />)
    wrapper.find('input').simulate('change', { target: { files: [filename] } })
    expect(onChange).toHaveBeenCalledWith(filename)
  })

  it('should fire onBlur', () => {
    const filename = '/a/path'
    const wrapper = shallow(<FileInput {...props} />)
    wrapper.find('input').simulate('blur', { target: { files: [filename] } })
    expect(onBlur).toHaveBeenCalledWith(filename)
  })
})
