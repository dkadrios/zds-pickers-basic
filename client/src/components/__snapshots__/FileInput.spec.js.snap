// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput tests renders correctly 1`] = `
<div>
  <input
    accept="text/plain"
    onBlur={[Function]}
    onChange={[Function]}
    type="file"
  />
  <p />
</div>
`;

exports[`FileInput tests should render a warning when touched 1`] = `
<div>
  <input
    accept="text/plain"
    onBlur={[Function]}
    onChange={[Function]}
    type="file"
  />
  <p>
    warning
  </p>
</div>
`;

exports[`FileInput tests should render an error when touched 1`] = `
<div>
  <input
    accept="text/plain"
    onBlur={[Function]}
    onChange={[Function]}
    type="file"
  />
  <p>
    error
  </p>
</div>
`;
